### PR TITLE
firewalk: init at 5.0

### PR DESCRIPTION
**Pull request by Tochiaha**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/firewalk/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    src = fetchurl {
+        url = "https://salsa.debian.org/pkg-security-team/firewalk/-/archive/debian/5.0-5/firewalk-debian-5.0-5.tar.gz";
+        hash = "sha256-luFTi4DnXbiNDI2uPgbtF4Y9m/SdxQBs+Dfdmck7mFU=";
+    };
+
+    buildInputs = [ libnet libpcap libdnet ];
+
+    meta = with lib; {
+        description = "Gateway ACL scanner";
+        homepage = "http://www.packetfactory.net/firewalk";
```
Comment (999eagle) [inline pkgs/applications/networking/firewalk/default.nix:22]: This page is dead.
```suggestion
        homepage = "http://packetfactory.openwall.net/projects/firewalk/";
```

**Review by 999eagle — CHANGES_REQUESTED** [review]
This is missing an entry like `firewalk = callPackage ../applications/networking/firewalk` in `pkgs/top-level/all-packages.nix` to actually call this package.

**Comment by 999eagle** [comment]
Relates to #81418.

File: pkgs/applications/networking/firewalk/default.nix (hunk 1 above)
Comment (Tochiaha) [reply pkgs/applications/networking/firewalk/default.nix]: changes accepted and made.

Comment (Tochiaha) [reply pkgs/applications/networking/firewalk/default.nix]: does an entry still apply after the recent changes?

Comment (natsukium) [reply pkgs/applications/networking/firewalk/default.nix]: Could you please read this comment and document? Thanks.

**Review by 999eagle — CHANGES_REQUESTED** [review]
You still should move your package to the [by-name hierarchy](https://github.com/NixOS/nixpkgs/tree/master/pkgs/by-name#readme) or, if you want to stick to the current category-based hierarchy, add this package to `pkgs/top-level/all-packages.nix` as detailed above.

**Comment by kirillrdy** [comment]
@Tochiaha can you please change commit message and PR title to `firewalk: init at 5.0`

File: pkgs/applications/networking/firewalk/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
+let
+  nixpkgs = import <nixpkgs> {  };
+
+in
+  with nixpkgs;
```
Comment (fabaff) [inline pkgs/applications/networking/firewalk/default.nix:5]: Please mention the inputs:

```suggestion
{ lib
, stdenv
, fetchurl
, libnet
...
}:
```

File: pkgs/applications/networking/firewalk/default.nix (hunk 1 above)
Comment (Tochiaha) [reply pkgs/applications/networking/firewalk/default.nix]: @natsukium, Thanks for the recommendation. Have i done the pkgs/by-name correctly?

File: pkgs/applications/networking/firewalk/default.nix (hunk 2 above)
Comment (Tochiaha) [reply pkgs/applications/networking/firewalk/default.nix:5]: Thanks, Done!

**Pull request closed without merging**